init/luks.go

```diff
@@ -298,6 +300,47 @@ func recoverTokenPassword(volumes chan *luks.Volume, d luks.Device, t luks.Token
 	info("password from %s token #%d does not match", t.Type, t.ID)
 }
 
+func recoverKeyfilePassword(volumes chan *luks.Volume, d luks.Device, checkSlots []int, mappingName string, keyfile string) {
+	var err error
+	var password []byte
+
+	// keyfile might be in the format /path:UUID=<DEV UUID> to indicate the keyfile lives on another device
+	parts := regexp.MustCompile("(?i):UUID=").Split(keyfile, 2)
+
+	if len(parts) == 1 {
+		password, err = os.ReadFile(parts[0])
+
+		if err != nil {
+			warning("reading password: %v", err)
+		}
+	} else {
+		// read password from device matching uuid
+		uuid, err := parseUUID(parts[1])
+		if err != nil {
+			warning("invalid UUID %s in rd.luks.key boot param: %s", uuid, keyfile)
+		} else {
+			// TODO: access path on uuid device, read password
+			warning("user wants keyfile from device %s, but I don't know how do do that", uuid)
```

Comment:
Looks like I made a typo here, "know how to do that" instead of "know how do do that"